tests/test_extract_model_langchain_openai.py

```diff
@@ -1,7 +1,9 @@
 from langchain_openai import AzureChatOpenAI, AzureOpenAI, ChatOpenAI, OpenAI
 import pytest
-
+import types
 from langfuse.callback import CallbackHandler
+from langchain.prompts import ChatPromptTemplate
+from langchain.schema import StrOutputParser
```

Comment:
FAILED tests/test_extract_model_langchain_openai.py::test_chain_streaming_llm_call_with_langchain_openai[gpt-3.5-turbo-model0] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_streaming_llm_call_with_langchain_openai[gpt-3.5-turbo-instruct-model1] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_streaming_llm_call_with_langchain_openai[gpt-4-0613-model2] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_streaming_llm_call_with_langchain_openai[gpt-3.5-turbo-model0] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_streaming_llm_call_with_langchain_openai[gpt-3.5-turbo-instruct-model1] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_streaming_llm_call_with_langchain_openai[gpt-4-0613-model2] - assert 2 == 4
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_invoke_llm_call_with_langchain_openai[gpt-3.5-turbo-model0] - UnboundLocalError: local variable 'res' referenced before assignment
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_invoke_llm_call_with_langchain_openai[gpt-3.5-turbo-instruct-model1] - UnboundLocalError: local variable 'res' referenced before assignment
FAILED tests/test_extract_model_langchain_openai.py::test_chain_async_invoke_llm_call_with_langchain_openai[gpt-4-0613-model2] - UnboundLocalError: local variable 'res' referenced before assignment
FAILED tests/test_extract_model_langchain_openai.py::test_chain_invoke_llm_call_with_langchain_openai[gpt-3.5-turbo-model0] - UnboundLocalError: local variable 'res' referenced before assignment
FAILED tests/test_extract_model_langchain_openai.py::test_chain_invoke_llm_call_with_langchain_openai[gpt-3.5-turbo-instruct-model1] - UnboundLocalError: local variable 'res' referenced before assignment
FAILED tests/test_extract_model_langchain_openai.py::test_chain_invoke_llm_call_with_langchain_openai[gpt-4-0613-model2] - UnboundLocalError: local variable 'res' referenced before assignment

Is it possible we have somewhat different setups? Tests seem to be failing here. Happy to jump on a call :)

Reply:
Maybe we can also move the test functions to a new file. This one is intended to ensure that we extract models correctly from a range of models. You were building more end to end tests where we test entire chains / calls